src/main/java/com/google/cloud/tools/app/DeployAction.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app;
+
+import com.google.cloud.tools.app.ProcessCaller.Tool;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Deploys an app to the server.
+ */
+public class DeployAction extends Action {
+
+  private static Set<Option> acceptedFlags = ImmutableSet.of(
+      Option.BUCKET,
+      Option.DOCKER_BUILD,
+      Option.FORCE,
+      Option.IMAGE_URL,
+      Option.PROMOTE,
+      Option.SERVER,
+      Option.VERSION
+  );
+  private File stagedDirectory;
+
+  /**
+   * Initialises all the necessary properties to deploy an application.
+   *
+   * @param stagedDirectoryLocation directory where staged application is located
+   * @param flags deployment flags
+   */
+  public DeployAction(String stagedDirectoryLocation, Map<Option, String> flags) {
+    super(flags);
+    Preconditions.checkNotNull(stagedDirectoryLocation);
```

Comment:
That's weird, is it in 3.0.0?
http://stackoverflow.com/questions/19030954/cant-find-nullable-inside-javax-annotation

On Wed, Mar 23, 2016 at 2:53 PM, João Martins <notifications@github.com>
wrote:

> In src/main/java/com/google/cloud/tools/app/DeployAction.java
> <https://github.com/GoogleCloudPlatform/gcp-tools-lib-for-java/pull/1#discussion_r57216596>
> :
>
> > +      Option.IMAGE_URL,
> > +      Option.PROMOTE,
> > +      Option.SERVER,
> > +      Option.VERSION
> > +  );
> > +  private File stagedDirectory;
> > +
> > +  /**
> > +   * Initialises all the necessary properties to deploy an application.
> > +   *
> > +   * @param stagedDirectoryLocation directory where staged application is located
> > +   * @param flags deployment flags
> > +   */
> > +  public DeployAction(String stagedDirectoryLocation, Map<Option, String> flags) {
> > +    super(flags);
> > +    Preconditions.checkNotNull(stagedDirectoryLocation);
>
> The following Maven artifact doesn't seem to contain Nullable / NonNull.
>
> <dependency>
>   <groupId>com.google.code.findbugs</groupId>
>   <artifactId>jsr305</artifactId>
>   <version>3.0.1</version>
> </dependency>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcp-tools-lib-for-java/pull/1/files/10d1211864218e1bd38d247da31c02ff9d402984#r57216596>
>